File: code3/fact.c
#include<stdio.h>

int fact(int num);

int main(void)
{
    int answer;
    answer = fact(5);
    printf("Factorial is: %d", answer);
    return 0;
}

int fact(int num)
{
    if(num == 0)
    {
        return 1;
    }
    else
    {
        return num * fact(num - 1);
    }
}

File: code3/fact.java
/*
 * factorial
 * author: Ashwek
 * github: @ashwek
 */

import java.lang.StringBuffer;

class factorial{

	public static int factorial(int num){
		if( num == 1){
			return 1;
		}
		else{
			return num * factorial(num-1);
		}
	}

	public static void main(String args[]){

		System.out.println("Factorial of 5 = " + factorial(5));

	}
}
